src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_io_net_stm32fx_nucleo_wifi.c

```diff
@@ -152,15 +163,18 @@ xi_bsp_io_net_state_t xi_bsp_io_net_close_socket( xi_bsp_socket_t* xi_socket )
     status               = wifi_socket_client_close( *xi_socket );
 
     /* free all buffers */
-
     xi_data_desc_t* head = xi_net_state.head;
+
     while ( NULL != head )
     {
         xi_data_desc_t* temp = head;
         head                 = head->__next;
         xi_free_desc( &temp );
     }
 
+    /* reset it to the default value after deleting all elements */
+    xi_net_state.head = NULL;
```

Comment:
Yeah there is a xi_list.h implementation - I think Milan was trying not add another dependency - I'll convert it to use it. That's a good idea @atigyi.